server/src/main/java/org/opensearch/cluster/coordination/JoinHelper.java

```diff
@@ -206,24 +230,55 @@ public ClusterTasksResult<JoinTaskExecutor.Task> execute(ClusterState currentSta
         transportService.registerRequestHandler(
             VALIDATE_JOIN_ACTION_NAME,
             ThreadPool.Names.GENERIC,
-            ValidateJoinRequest::new,
+            BytesTransportRequest::new,
             (request, channel, task) -> {
+                handleValidateJoinRequest(currentStateSupplier, joinValidators, request);
+                channel.sendResponse(Empty.INSTANCE);
+            }
+        );
+    }
+
+    private void handleValidateJoinRequest(
```

Comment:
This logic would need to account for the input/ output stream version as there could be mixed mode clusters
1. Leader sends old cluster state (uncompressed) while node n1 expects uncompressed state -- No issue
2. Leader sends old cluster state (uncompressed) while node n1 expects compressed state  --- It would fail
3. Leader sends compressed state while node n1 expects compressed state  -- No Issue
4. Leader sends compressed state while node n1 expects uncompressed state.   --- It would fail (though this case, OpenSearch doesn't guarantee)

Code needs to handle use case for 2 & 4

Reply:
Adding a different action name for sending compressed state as well as adding handler separately. Thanks I missed this.